rust/otap-dataflow/crates/otap/src/pdata.rs

```diff
@@ -0,0 +1,310 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
A comment should be added to this module. I’m especially thinking of an ASCII diagram that would represent OtapPdata and its different states.

Reply:
I added a comment to explain the different representations.

I wasn't sure what we wanted as an ascii diagram. I added one that explains the conversions, but happy to tweak this if you were looking for something different specifically